contracts/Shortfall/Shortfall.sol

```diff
@@ -222,7 +262,12 @@ contract Shortfall is Ownable2StepUpgradeable, AccessControlledV8, ReentrancyGua
 
         auction.highestBidder = msg.sender;
         auction.highestBidBps = bidBps;
-        auction.highestBidBlock = block.number;
+
+        if (Toggle == Mode.BLOCK_NUMBER) {
+            auction.highestBidBlock = block.number;
+        } else {
+            auction.highestBidBlock = block.timestamp;
+        }
```

Comment:
If we follow the inheritance approach, we could have a virtual function `getBlockNumberOrTimestamp()`, and implement it specifically on each specific contract:

```
auction.highestBidBlock = getBlockNumberOrTimestamp();
...
function getBlockNumberOrTimestamp() virtual;
...
contract BlockNumberBasedShortfall {
  function getBlockNumberOrTimestamp() { return block.number; }
}
...
contract TimeBasedShortfall {
  function getBlockNumberOrTimestamp() { return block.timestamp; }
}
```